### PR TITLE
Jetpack Connection: Add inline comments when whole input is processed or nonce verification is missing

### PR DESCRIPTION
**Pull request by fgiannar**
We are adding inline comments in the Connection package to explain cases where the whole `$_POST` input is used.
We are also refactoring `setup_xmlrpc_handlers` in `Manager` class to avoid using the whole `$_POST` input.

## Proposed changes:

* `Jetpack_Signature::sign_current_request`: We need the whole `$_POST` in order to verify a cryptographic signature of the post data.
* `Automattic\Jetpack\Connection\Manager::internal_verify_xml_rpc_signature`: We need the whole `$_POST` in order to verify a cryptographic signature of the post data.
* `Automattic\Jetpack\Connection\Manager::setup_xmlrpc_handlers`: Deprecate the `$request_params` method argument and check for `$_GET['for']` and `$_GET['jetpack']` inside the method instead. I've audited the usage of this method and confirmed it's used only by the same class.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p9dueE-8ca-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

- For the inline comments: Code Review: Do you believe we need further explanation?
- For the change in `setup_xmlrpc_handlers`: I believe the best way to test this change (apart from the code review) would be to use the JP Debugger and confirm the same results for a given site with/without the PR applied. The JP Debugger will make a bunch of REST and XML-RPC requests to the site with/without authentication too. Also make sure to set the site to use the alternate XML-RPC url in order to set the `comms` param is properly checked too.

### CODE REVIEW

File: projects/packages/connection/legacy/class-jetpack-signature.php [hunk 1]
```diff
@@ -93,9 +93,9 @@ public function sign_current_request( $override = array() ) {
 			// Convert the $_POST to the body, if the body was empty. This is how arrays are hashed
 			// and encoded on the Jetpack side.
 			if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
-				// phpcs:ignore WordPress.Security.NonceVerification.Missing
+				// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Used to generate a cryptographic signature of the post data.
```
Comment (anomiex) [inline projects/packages/connection/legacy/class-jetpack-signature.php]:
```suggestion
				// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Used to generate a cryptographic signature of the post data. Not actually using any of it here.
```

File: projects/packages/connection/src/class-manager.php [hunk 2]
```diff
@@ -455,9 +454,9 @@ private function internal_verify_xml_rpc_signature() {
 		}
 
 		$jetpack_signature = new \Jetpack_Signature( $token->secret, (int) \Jetpack_Options::get_option( 'time_diff' ) );
-		// phpcs:disable WordPress.Security.NonceVerification.Missing
+		// phpcs:disable WordPress.Security.NonceVerification.Missing -- Used to verify a cryptographic signature of the post data.
```
Comment (anomiex) [inline projects/packages/connection/src/class-manager.php]:
```suggestion
		// phpcs:disable WordPress.Security.NonceVerification.Missing -- Used to verify a cryptographic signature of the post data. Also a nonce is verified later in the function.
```

**Review by anomiex — COMMENTED** [review]
LGTM. A few suggestions inline.

**Review by anomiex — DISMISSED** [review]
LGTM. Haven't tested though.

**Review by darssen — APPROVED** [review]
Did test provided instructions for `setup_xmlrpc_handlers` and things looked good.

**Pull request merged**